### ISSUE TITLE
Can we upgrade NumPy version from 1.22 to 1.24 or more for Python 3.11?

### ISSUE THREAD

**Issue by rinarakaki** [skvark/opencv-python#835]
### Expected behaviour

`numpy==1.22` emits warning: `RuntimeWarning: NumPy 1.22.0 may not yet support Python 3.11.`

### Actual behaviour

NumPy's Python 3.11 support begins from version 1.24.

### Steps to reproduce

- `pip3 install opencv-python`

##### Issue submission checklist

 - [x] This is not a generic OpenCV usage question (looking for help for coding, other usage questions, homework etc.)
   
 - [x] I have read the README of this repository and understand that this repository provides only an automated build toolchain for OpenCV Python packages (there is no actual OpenCV code here)
   
 - [x] The issue is related to the build scripts in this repository, to the pre-built binaries or is a feature request (such as "please enable this additional dependency")
 - [x] I'm using the latest version of ``opencv-python``

**Comment by tomsweeting** [comment]
@rnarkk I also encountered this issue today, but instead of installing via pip I was trying to build from source for Python 3.11 with gstreamer support.

I was able to compile by setting the required numpy version 1.24.3, so I have put in a small PR to fix the issue. If you urgently need this to work with Python 3.11, it is possible to clone the repo and build manually as I did by making the same changes.
